lib/Request.js

```diff
@@ -310,3 +313,89 @@ Request.prototype.propertyError = function(property, errorCode, httpStatusCode,
         });
     }.bind(this));
 };
+
+Request.prototype.getStream = function() {
+    if (null === this._stream) {
+        this._stream = new Busboy({
+            headers: this._connection.raw.req.headers
+        });
+        this._connection.raw.req.pipe(this._stream);
+    }
+    return this._stream;
+}
+
+Request.prototype.getUpload = function(field) {
+    return new Promise(function(resolve, reject) {
+        var stream = this.getStream();
+        var found = false;
+
+        stream.on('file', function(uploadedFieldName, file, filename, encoding, mimetype) {
+            if (field != uploadedFieldName) {
+                // Got an upload, but not for the field name we are looking for.
+                // Skip this upload, maybe the next one will be under the desired field name.
+                return;
+            }
+
+            found = true;
+
+            // Aggregrate the data as it is streamed in
+            var chunks = [];
+
+            file.on('data', function(data) {
+                chunks.push(data)
+            });
+
+            file.on('end', function() {
+                resolve({
+                    buffer: Buffer.concat(chunks),
+                    fieldname: uploadedFieldName,
+                    encoding: encoding,
+                    mimetype: mimetype,
+                    filename: filename
+                });
+            });
+        });
+
+        stream.on('finish', function() {
+            if (!found) {
+                reject(new Error('Did not find uploaded file under field name ' + field));
```

Comment:
No, I don't think we should throw an error for skipped fields. By using the design as implemented, we can fetch two uploads without having to deal with error juggling:

    return Promise.all([
      Request.getUpload('foo'),
      Request.getUpload('bar')
    ])
    .then(function(uploads) {
      // Do something with each upload
    })
    .catch(function(error) {
      // One or both uploads missing, or some other error like wrong content type
    });